packages/ag-charts-community/src/api/preset/sparkline.ts

```diff
@@ -182,18 +186,21 @@ export function sparklineDataPreset(data: any[] | undefined): {
             const mappedData = data.map((y, x) => ({ x, y }));
             return { data: mappedData, series: [{ xKey: 'x', yKey: 'y' }] };
         } else if (Array.isArray(firstItem)) {
-            const mappedData = data.map(([x, y]) => ({ x, y }));
+            const mappedData = data.filter((item) => !!item).map(([x, y]) => ({ x, y }));
             return { data: mappedData, series: [{ xKey: 'x', yKey: 'y' }] };
         }
     }
-
     return { data };
 }
 
 function axisPreset(
     opts: AgSparklineAxisOptions | undefined,
     defaultType: AgCartesianAxisOptions['type']
 ): AgCartesianAxisOptions {
+    opts = {
+        ...opts,
+        type: opts?.type ?? defaultType,
+    } as any as AgSparklineAxisOptions;
```

Comment:
Presuming was can avoid this if `axis.type` is required?

Reply:
It currently defaults to 'category' in prod, so it's optional from the user perspective. There's a comment higher up where we're discussing al alternative approach

Reply:
Removed, type is now required